### PR TITLE
Add design document about handing some MAX defines

### PR DESCRIPTION
**Pull request by t8m**
This design document proposes way(s) to handle some of the MAX defines in the public headers.

### CODE REVIEW

**Review by tom-cosgrove-arm — APPROVED** [review]
Nice analysis, thanks

File: doc/designs/handling-some-max-defines.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+### EVP_MAX_MD_SIZE
+
+**Current value:** 64
+
+It is unlikely we will see longer than 512 bit hashes any time soon.
+XOF functions do not count and the XOF output length is not and should
+not be limited by this value.
+
+It is widely used throughout the codebase and by 3rd party applications.
+
```
Comment (nhorman) [inline doc/designs/handling-some-max-defines.md:36]: Whats the proposed solution here?  convert users of this macro to EVP_MD_get_size and remove the macro for 4.0?

**Comment by beldmit** [comment]
Nice analysis!

**Comment by t8m** [comment]
**OTC: Added hold to discuss this before it is merged.**

**Comment by t8m** [comment]
@beldmit could you please add formal review approval.

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to the master branch. Thank you for the reviews.

**Pull request closed without merging**